### PR TITLE
Move rtc packages

### PR DESCRIPTION
**Pull request by hbcarlos**
This PR moves RTC packages from JupyterLab to Jupyter_server_ydoc keeping the git history. It just creates a folder `packages` with the RTC packages without adding lerna or creating any extension. I will create the extension and set up the repository in another PR.

### CODE REVIEW

File: packages/docprovider/tsconfig.test.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "extends": "../../tsconfigbase.test",
```
Comment (fcollonval) [inline packages/docprovider/tsconfig.test.json:2]: You need to bring that file in.

File: packages/docprovider/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "extends": "../../tsconfigbase",
```
Comment (fcollonval) [inline packages/docprovider/tsconfig.json:2]: You need to bring that file in.

File: packages/docprovider/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  },
+  "include": ["src/*"],
+  "references": [
+    {
+      "path": "../coreutils"
+    },
+    {
+      "path": "../services"
+    }
+  ]
```
Comment (fcollonval) [inline packages/docprovider/tsconfig.json:15]:
```suggestion
```

File: packages/docprovider/tsconfig.test.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    {
+      "path": "../services"
+    },
+    {
+      "path": "."
+    },
+    {
+      "path": "../testing"
+    }
+  ]
```
Comment (fcollonval) [inline packages/docprovider/tsconfig.test.json:17]:
```suggestion
```

File: packages/docprovider/package.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
+{
+  "name": "@jupyterlab/docprovider",
+  "version": "4.0.0-alpha.18",
+  "description": "JupyterLab - Document Provider",
+  "homepage": "https://github.com/jupyterlab/jupyterlab",
```
Comment (fcollonval) [inline packages/docprovider/package.json:5]: Could you do a search-replace to correct the URL?

File: packages/collaboration/typedoc.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "out": "../../docs/api/collaboration",
+  "theme": "../../typedoc-theme"
```
Comment (fcollonval) [inline packages/collaboration/typedoc.json:3]: You need to bring this in too.

File: packages/collaboration/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    {
+      "path": "../coreutils"
+    },
+    {
+      "path": "../services"
+    },
+    {
+      "path": "../ui-components"
+    }
+  ]
```
Comment (fcollonval) [inline packages/collaboration/tsconfig.json:21]:
```suggestion
```

File: packages/collaboration/style/index.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+/* This file was auto-generated by ensurePackage() in @jupyterlab/buildutils */
+import '@lumino/widgets/style/index.js';
+import '@jupyterlab/ui-components/style/index.js';
+import '@jupyterlab/apputils/style/index.js';
+
```
Comment (fcollonval) [inline packages/collaboration/style/index.js:10]:
```suggestion
```

File: packages/collaboration/style/index.css [hunk 9]
```diff
@@ -0,0 +1,11 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+/* This file was auto-generated by ensurePackage() in @jupyterlab/buildutils */
+@import url('~@lumino/widgets/style/index.css');
+@import url('~@jupyterlab/ui-components/style/index.css');
+@import url('~@jupyterlab/apputils/style/index.css');
+
```
Comment (fcollonval) [inline packages/collaboration/style/index.css:10]:
```suggestion
```

File: packages/collaboration-extension/tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    {
+      "path": "../statedb"
+    },
+    {
+      "path": "../translation"
+    },
+    {
+      "path": "../ui-components"
+    }
+  ]
```
Comment (fcollonval) [inline packages/collaboration-extension/tsconfig.json:33]:
```suggestion
```

File: packages/collaboration-extension/style/index.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+/* This file was auto-generated by ensurePackage() in @jupyterlab/buildutils */
+import '@lumino/widgets/style/index.js';
+import '@jupyterlab/ui-components/style/index.js';
+import '@jupyterlab/apputils/style/index.js';
+import '@jupyterlab/application/style/index.js';
+import '@jupyterlab/collaboration/style/index.js';
+
```
Comment (fcollonval) [inline packages/collaboration-extension/style/index.js:12]:
```suggestion
```

File: packages/collaboration-extension/style/index.css [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+/* This file was auto-generated by ensurePackage() in @jupyterlab/buildutils */
+@import url('~@lumino/widgets/style/index.css');
+@import url('~@jupyterlab/ui-components/style/index.css');
+@import url('~@jupyterlab/apputils/style/index.css');
+@import url('~@jupyterlab/application/style/index.css');
+@import url('~@jupyterlab/collaboration/style/index.css');
+
```
Comment (fcollonval) [inline packages/collaboration-extension/style/index.css:12]:
```suggestion
```

File: packages/collaboration-extension/style/base.css [hunk 13]
```diff
@@ -0,0 +1,4 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
```
Comment (fcollonval) [inline packages/collaboration-extension/style/base.css:4]: I guess we can drop this.

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @hbcarlos

Could you add the top `package.json` and the associate GitHub action?

> In particular the linter configuration.

**Comment by hbcarlos** [comment]
Hi @fcollonval.

I know we need to remove some files and bring others from JupyterLab. The idea of this PR is only to get the git history from JupyterLab.

The configuration of a monorepo with multiple JupyterLab extensions is slightly different than the JupyterLab repo. I would like to do all that work in another PR to have a cleaner git history and make it easier to review.

The work done in this PR is in an isolated folder that doesn't interfere with this extension.

**Comment by fcollonval** [comment]
Ok then let's move on fast with small PRs to first restore the CI.

**Review by fcollonval — APPROVED** [review]

**Comment by hbcarlos** [comment]
Thanks!!

**Pull request merged**